// grunt-jasmine-node-coverage-validation/Gruntfile.js
module.exports = function (grunt) {
    'use strict';

    grunt.initConfig({
      jshint: {
        options: {
          curly: true,
          eqeqeq: true,
          immed: true,
          latedef: true,
          newcap: true,
          noarg: true,
          sub: true,
          undef: true,
          boss: true,
          eqnull: true,
          node: true,
          es5: true
        },
        all: ['tasks/**/*.js']
      },
      jasmine_node: {
        coverage: {
          options : {
            branches : 100 ,
            functions: 1000,
            statements:100,
            lines:1000
          }
        },
        options: {
          forceExit: true,
          match: '.',
          matchall: false,
          extensions: 'js',
          specNameMatcher: 'spec',
          junitreport: {
            report: false,
            savePath : "./build/reports/jasmine/",
            useDotNotation: true,
            consolidate: true
          }
        },
        all: ['spec/']
      }
    });

    grunt.loadNpmTasks('grunt-contrib-jshint');
    grunt.loadTasks('tasks');

    //grunt.registerTask('default', ['jshint' ,'jasmine_node']);
	
   grunt.registerTask('default', ['jasmine_node']);


};
